tests/testWarpExposure.py

```diff
@@ -432,6 +432,47 @@ def testNonIcrs(self):
                 self.assertImagesAlmostEqual(afwWarpedImage, swarpedImage,
                                              skipMask=noDataMaskArr, rtol=rtol, atol=atol)
 
+    @unittest.skipIf(afwdataDir is None, "afwdata not setup")
+    def testTransformBasedWarp(self):
+        """Test warping using Transform<Point2Endpoint, Point2Endpoint>
+        """
+        for interpLength in (0, 1, 2, 4):
+            kernelName = "lanczos3"
+            rtol = 4e-5
+            atol = 1e-2
+            warpingControl = afwMath.WarpingControl(
+                warpingKernelName = kernelName,
+                interpLength = interpLength,
+            )
+
+            originalExposure = afwImage.ExposureF(originalExposurePath)
+            originalMetadata = afwImage.DecoratedImageF(originalExposurePath).getMetadata()
+            originalSkyWcs = afwGeom.SkyWcs(originalMetadata)
+
+            swarpedImageName = "medswarp1%s.fits" % (kernelName,)
+            swarpedImagePath = os.path.join(dataDir, swarpedImageName)
+            swarpedDecoratedImage = afwImage.DecoratedImageF(swarpedImagePath)
+            swarpedImage = swarpedDecoratedImage.getImage()
+
+            swarpedMetadata = swarpedDecoratedImage.getMetadata()
+            warpedSkyWcs = afwGeom.SkyWcs(swarpedMetadata)
+
+            # original image is source, warped image is destination
+            srcFromDest = originalSkyWcs.getInverse().of(warpedSkyWcs)
```

Comment:
If we want it to be less confusing I fear some syntactic sugar may be required. I'm reluctant to fatten the API of `SkyWcs` until we are further along, preferring to document what's going on, especially since (as I say above) few users will ever need to do this.